src/transaction_builder.js

```diff
@@ -394,34 +414,43 @@ TransactionBuilder.prototype.sign = function (index, keyPair, redeemScript, hash
         if (!bufferEquals(scriptHash, bcrypto.hash160(redeemScript))) throw new Error('RedeemScript does not match ' + scriptHash.toString('hex'))
       }
 
-      var scriptType = bscript.classifyOutput(redeemScript)
-      var redeemScriptChunks = bscript.decompile(redeemScript)
-      var pubKeys
+      var pubKeys, pkh1, pkh2
 
-      switch (scriptType) {
-        case 'multisig':
-          pubKeys = redeemScriptChunks.slice(1, -2)
+      var redeemScriptType
 
-          break
+      var processScript = function (redeemScript) {
```

Comment:
moved into a function so that we can call it again for nested scripts (P2SH[ P2WSH[ P2PKH / multsig ] ]